src/features/WriteStoriesModal/WriteStoriesModal.tsx

```diff
@@ -2,27 +2,272 @@ import React from "react";
+  Main,
+  ModalContent,
+  SpanHighlight,
+} from "./WriteStoriesModal.styled";
+import { SyntaxHighlighter } from "../../components/SyntaxHighlighter/SyntaxHighlighter";
+import { List } from "../../components/List/List";
+import { ListItem } from "../../components/List/ListItem/ListItem";
+import { useGetButtonPath } from "./hooks/useGetButtonPath";
+import { useGetWarningButtonStatus } from "./hooks/useGetWarningButtonStatus";
+import { useGetBackdropBoundary } from "./hooks/useGetBackdropBoundary";
+import titleSidebarImg from "./assets/01-title-sidebar.png";
+import storyNameSidebarImg from "./assets/02-story-name-sidebar.png";
+import argsImg from "./assets/03-args.png";
+
+import dataJavascript from "./code/javascript";
+import dataTypescript from "./code/typescript";
+import dataTypescriptNextjs from "./code/nextjs-typescript";
+import { useGetProject } from "./hooks/useGetFrameworkName";
+import { API, AddonStore } from "@storybook/manager-api";
+
+export function WriteStoriesModal({
+  onFinish,
+  api,
+  addonsStore,
+}: {
+  onFinish: () => void;
+  api: API;
+  addonsStore: AddonStore;
+}) {
+  const [step, setStep] = React.useState<
+    "imports" | "meta" | "story" | "args" | "customStory"
+  >("imports");
+
+  const [isWarningStoryCopied, setWarningStoryCopied] = React.useState(false);
+
+  const buttonPath = useGetButtonPath();
+  const doesWarningButtonExist = useGetWarningButtonStatus(
+    step === "customStory",
+    api,
+    addonsStore
+  );
+  const backdropBoundary = useGetBackdropBoundary(
+    "syntax-highlighter-backdrop",
+    step === "customStory"
+  );
+
+  const project = useGetProject();
+  const isJavascript = project?.data?.language === "javascript";
+
+  const data = isJavascript
+    ? dataJavascript
+    : project?.data?.framework.name === "@storybook/nextjs"
+    ? dataTypescriptNextjs
+    : dataTypescript;
+
+  const copyWarningStory = () => {
+    const warningContent = data[4][0].content;
+    navigator.clipboard.writeText(
+      warningContent.replace("// Copy the code below", "")
+    );
+    setWarningStoryCopied(true);
+  };
 
-export function WriteStoriesModal({ onFinish }: { onFinish: () => void }) {
   return (
-    <Modal width={540} defaultOpen>
-      {({ Title, Description }) => (
-        <ModalContentWrapper data-chromatic="ignore">
-          <StorybookLogo />
-          <Title style={{ marginTop: 20 }}>
-            Create your first story (WORK IN PROGRESS)
-          </Title>
-          <Description>
-            Now it's your turn. <br />
-            See how easy it is to create your first story by following these
-            steps below.
-          </Description>
-          <Button style={{ marginTop: 4 }} onClick={onFinish}>
-            Go to story
-          </Button>
-        </ModalContentWrapper>
+    <Modal width={738} defaultOpen>
+      {({ Title, Description: DefaultDescription, Close }) => (
+        <ModalContent>
+          <div style={{ maxHeight: "450px" }}>
```

Comment:
Should this contain a minHeight too, so that the layout jump is not too much?

Reply:
oh yeah! I forgot about it. I will take a look!